vmicore/src/vmi/InterruptEvent.cpp

```diff
@@ -89,33 +91,32 @@ void InterruptEvent::setupVmiInterruptEvent()
 
 void InterruptEvent::enableEvent()
 {
+#if defined(X86_64)
     vmiInterface->write8PA(targetPA, INT3_BREAKPOINT);
+#elif defined(ARM64)
+    vmiInterface->write32PA(targetPA, BRK64_BREAKPOINT);
+#endif
     logger->debug("Enabled interrupt event", {logfield::create("targetPA", targetPAString)});
 }
 
 void InterruptEvent::disableEvent()
 {
+#if defined(X86_64)
     vmiInterface->write8PA(targetPA, originalValue);
+#elif defined(ARM64)
+    vmiInterface->write32PA(targetPA, originalValue);
+#endif
     logger->debug("Disabled interrupt event", {logfield::create("targetPA", targetPAString)});
 }
 
-uint64_t InterruptEvent::getRcx()
-{
-    return event.x86_regs->rcx;
-}
-
-uint64_t InterruptEvent::getRdi()
-{
-    return event.x86_regs->rdi;
-}
-
-uint64_t InterruptEvent::getR8()
+registers_t* InterruptEvent::getRegisters()
 {
-    return event.x86_regs->r8;
+    return (registers_t*)event.x86_regs;
```

Comment:
I think this is problematic since it breaks abstraction from libvmi types which was kinda the goal here. Afaik @K-Mayer and @cakeless are currently working on a PR that will expose the `InterruptEvent` to plugins.

Reply:
I like the `getRegisters()` approach though. It would probably make sense to create an abstract `IRegisters` class and implement that. (yes, I know it's a lot of boiler plate, I am not happy about that either).